Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,560 @@
+                                                                             Real, Real, Real, Real,
+                                                                             Real, Real, Real, Real>
+            {
+                const amrex::Real w  = p.rdata(PIdx::w);
+                const amrex::Real x = p.pos(0);
+                const amrex::Real y = p.pos(1);
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uy = p.rdata(PIdx::uy);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                const amrex::Real thetay = std::atan2(uy, uz);
+                return {w, w*x, w*x*x, w*y, w*y*y,
+                        thetax, w*thetax, w*thetax*thetax, thetax,
+                        thetay, w*thetay, w*thetay*thetay, thetay};
+            },
+            reduce_ops);
+
+        w_tot  = amrex::get<0>(r);
+        x_ave  = amrex::get<1>(r);
+        x2_ave = amrex::get<2>(r);
+        y_ave  = amrex::get<3>(r);
+        y2_ave = amrex::get<4>(r);
+        thetax_min  = amrex::get<5>(r);
+        thetax_ave  = amrex::get<6>(r);
+        thetax2_ave = amrex::get<7>(r);
+        thetax_max  = amrex::get<8>(r);
+        thetay_min  = amrex::get<9>(r);
+        thetay_ave  = amrex::get<10>(r);
+        thetay2_ave = amrex::get<11>(r);
+        thetay_max  = amrex::get<12>(r);
+
+        amrex::ParallelDescriptor::ReduceRealSum({w_tot, x_ave, x2_ave, y_ave, y2_ave,
+                                                  thetax_ave, thetax2_ave, thetay_ave, thetay2_ave},
+                                                  amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMin({thetax_min, thetay_min}, amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMax({thetax_max, thetay_max}, amrex::ParallelDescriptor::IOProcessorNumber());
+
+
+        if (w_tot > std::numeric_limits<Real>::min())
+        {
+            x_ave = x_ave / w_tot;
+            x2_ave = x2_ave / w_tot;
+            y_ave = y_ave / w_tot;
+            y2_ave = y2_ave / w_tot;
+            thetax_ave = thetax_ave / w_tot;
+            thetax2_ave = thetax2_ave / w_tot;
+            thetay_ave = thetay_ave / w_tot;
+            thetay2_ave = thetay2_ave / w_tot;
+        }
+        else
+        {
+            x_ave = 0.0_rt;
+            x2_ave = 0.0_rt;
+            y_ave = 0.0_rt;
+            y2_ave = 0.0_rt;
+            thetax_ave = 0.0_rt;
+            thetax2_ave = 0.0_rt;
+            thetay_ave = 0.0_rt;
+            thetay2_ave = 0.0_rt;
+        }
+
+        // x_std, y_std, thetax_std, thetay_std
+        amrex::Real x_std = 0.0_rt;
+        amrex::Real y_std = 0.0_rt;
+        amrex::Real thetax_std = 0.0_rt;
+        amrex::Real thetay_std = 0.0_rt;
+
+        x_std = std::sqrt(x2_ave - x_ave*x_ave);
```

Comment:
I've changed my mind on this. The tests I did showed that doing it this way does break down if the average is much larger than the std. I now think doing two loops is better and more robust.

Reply:
Oh ok, I haven't tested it yet in fact. 
@ax3l, what do you think?